openqa-label-known-issues

```diff
@@ -134,14 +134,16 @@ investigate_issue() {
     # against even in case when autoinst-log.txt is missing the details, e.g.
     # see https://progress.opensuse.org/issues/69178
     echo "$reason" >> "$out"
-    if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]]; then
+    # Control flow when there is a curl issue but still want to process job due to reason var
+    # even if autoinst-log.txt not found but assume is missing.
+    [[ -n $reason ]] && [[ $curl_output = "404" ]] && [[ $reason != null ]] \
+        && is_unreachachle_but_has_reason=1 \
+        || is_unreachachle_but_has_reason=0
+    if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]] && [[ ! $is_unreachachle_but_has_reason ]]; then
```

Comment:
This is most probably not what you want. Your `[[ ! "$is_unreachachle_but_has_reason ]]`condition would be always false. You might want something like.

```suggestion
    [[ -n "$reason" ]] && [[ "$curl_output" = "404" ]] && [[ "$reason" != null ]] && is_unreachachle_but_has_reason=1
    if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]] && [[ -z "$is_unreachachle_but_has_reason" ]]; then
```

Note: I suppose _null_ is just a plain string returned from the `jq`, right?

Reply:
I think that null is indeed a plain string.
I guess I have to quote all the variables. 

But I dont get why `-z` or `condition would be always false`?! if you run the test cases, it is not always `false`

Reply:
You have
```   
 [[ -n $reason ]] && [[ $curl_output = "404" ]] && [[ $reason != null ]] \
        && is_unreachachle_but_has_reason=1 \
        || is_unreachachle_but_has_reason=0
```
which will either set `is_unreachachle_but_has_reason` to `1` or `0`, right? The following condition `[[ ! $is_unreachachle_but_has_reason ]]` is functionally equivalent of `[[ -! -n $is_unreachachle_but_has_reason ]]` which can be seen as `[[ -z $is_unreachachle_but_has_reason ]]`, in other other words will evaluate to false if the string has non-zero length. In your case it is `0` or `1` which is always a non-zero length string.

Reply:
https://github.com/os-autoinst/scripts/compare/1c803f6248b784d98ee87dc07abc12f398876b05..1208b09e6428bb97e65be90680b9cb8b2c3c9ed9 added the quotes

Reply:
Done https://github.com/os-autoinst/scripts/compare/1208b09e6428bb97e65be90680b9cb8b2c3c9ed9..f6dfb2e88b82f1d9659759597095521048e4040b